module/VuFindTheme/src/VuFindTheme/View/Helper/Slot.php

```diff
@@ -0,0 +1,209 @@
+     *
+     * @param string $name Name of target block for action
+     *
+     * @return Slot
+     */
+    public function __invoke($name)
+    {
+        $this->stack[] = $name;
+        return $this;
+    }
+
+    /**
+     * Helper function to return blocks with prepends and appends
+     *
+     * @param string $name Name of target block for action
+     *
+     * @return string|any
+     */
+    protected function build($name)
+    {
+        $pre = $this->blockPrepends[$name] ?? [];
+        $block = $this->blocks[$name] ?? '';
+        $post = $this->blockAppends[$name] ?? [];
+        if (!empty($pre) || !empty($post)) {
+            // TODO: Warn about non-string blocks
+            return implode(' ', $pre) . $block . implode(' ', $post);
+        }
+        if (empty($block)) {
+            return null;
+        }
+        return $block;
+    }
+
+    /**
+     * Get current value of slot. Returns null if unset.
+     *
+     * @return string|null
+     */
+    public function get()
+    {
+        $name = array_pop($this->stack);
+        return $this->build($name);
+    }
+
+    /**
+     * Set current value of slot but only if unset.
+     *
+     * @param any $value Value to override if unset
+     *
+     * @return string|null
+     */
+    public function set($value)
+    {
+        $name = array_pop($this->stack);
+        if (!isset($this->blocks[$name])) {
+            $this->blocks[$name] = $value;
+        }
+        return $this->build($name);
+    }
+
+    /**
+     * Add string to list of block prepends.
+     *
+     * @param string $value Value to override if unset
+     *
+     * @return string
+     */
+    public function prepend($value)
+    {
+        $name = array_pop($this->stack);
+        if (!isset($this->blockPrepends[$name])) {
+            $this->blockPrepends[$name] = [$value];
+        } else {
+            array_unshift($this->blockPrepends[$name], $value);
+        }
+        return $this->build($name);
+    }
+
+    /**
+     * Add string to list of block appends.
+     *
+     * @param string $value Value to override if unset
+     *
+     * @return string
+     */
+    public function append($value)
+    {
+        $name = array_pop($this->stack);
+        if (!isset($this->blockAppends[$name])) {
+            $this->blockAppends[$name] = [$value];
+        } else {
+            array_push($this->blockAppends[$name], $value);
+        }
+        return $this->build($name);
+    }
+
+    /**
+     * Starts a buffer capture to override the value of a block.
+     *
+     * @return void
+     */
+    public function start()
+    {
+        array_pop($this->stack);
+        ob_start();
+    }
+
+    /**
+     * End a buffer capture to override the value of a block. Returns slot value.
+     *
+     * @param string $method SET/PREPEND/APPEND for where this buffer should be saved
+     *
+     * @return string|any
+     */
+    public function end($method = 'SET')
+    {
+        $ret;
```

Comment:
I think you're missing something here.

Reply:
Maybe I'm confusing for JS scoping, but `$ret` is set in the if statements right below it and I wanted access to it.

Reply:
In PHP, the statement `$ret;` by itself is just going to throw an "Undefined variable $ret" notice. I'm assuming that you wanted to actually initialize this to a value.